modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotPartitionsVerifyHandler.java

```diff
@@ -106,8 +108,26 @@ public SnapshotPartitionsVerifyHandler(GridCacheSharedContext<?, ?> cctx) {
 
         SnapshotMetadata meta = opCtx.metadata();
 
-        Set<Integer> grps = F.isEmpty(opCtx.groups()) ? new HashSet<>(meta.partitions().keySet()) :
-            opCtx.groups().stream().map(CU::cacheId).collect(Collectors.toSet());
+        Set<Integer> grps = F.isEmpty(opCtx.groups())
+            ? new HashSet<>(meta.partitions().keySet())
+            : opCtx.groups().stream().map(CU::cacheId).collect(Collectors.toSet());
+
+        grps = grps.stream().filter(grp -> {
+            if (grp == MetaStorage.METASTORAGE_CACHE_ID)
+                return true;
+
+            CacheGroupDescriptor grpDesc = cctx.kernalContext().cache().cacheGroupDescriptor(grp);
+
+            if (grpDesc == null)
+                return true; // Restore. Skip filter.
+
+            IgnitePredicate<ClusterNode> filter = grpDesc.config().getNodeFilter();
+
+            return filter == null || CU.affinityNode(
```

Comment:
Think we don't need line separators here

Reply:
Fixed.